react/src/components/atoms/forms/InputNumber/index.js

```diff
@@ -52,45 +52,34 @@ const NumberInput = (props) => {
               errorMsg: ''
             };
           };
-          const hasProperty = (obj, property) => Object.prototype.hasOwnProperty.call(obj, property) && !is.nil(obj[property]);
+
+          const hasNumberProperty = (obj, property) => Object.prototype.hasOwnProperty.call(obj, property) && is.number(obj[property]);
 
           const handleOnBlur = (e) => {
             e.persist();
             const inputEl = ref.current;
-            let newValue = Number(inputEl.value);
-            if ((hasProperty(props, 'max') && newValue > props.max) || (hasProperty(props, 'min') && newValue < props.min)) {
-              if (hasProperty(props, 'max') && newValue > props.max) {
-                newValue = props.max;
-              }
-              if (hasProperty(props, 'min') && newValue < props.min) {
-                newValue = props.min;
-              }
+            let newValue = inputEl.value ? Number(inputEl.value) : inputEl.value;
```

Comment:
This wouldn't work for '0'. Maybe this?
```suggestion
            let newValue = is.number(inputEl.value) ? Number(inputEl.value) : inputEl.value;
```

Reply:
inputEl.value is a string so this is fine
if it was 0 (a number and not a string), it would still be assigned to 0.

Reply:
If it was 0 it would be a number and newValue would be 0, if it was a string "0", than it would be case to a number 0 - so i think this would work